IPython/core/displayhook.py

```diff
@@ -83,9 +84,19 @@ def check_for_underscore(self):
     def quiet(self):
         """Should we silence the display hook because of ';'?"""
         # do not print output if input ends in ';'
+ 
+	cell = self.shell.history_manager.input_hist_parsed[self.prompt_count]
+        sio = io.StringIO(cell)
+        tokens = list(tokenize.generate_tokens(sio.readline))
+ 
         try:
-            cell = self.shell.history_manager.input_hist_parsed[self.prompt_count]
-            return cell.rstrip().endswith(';')
+    	    for token in reversed(tokens):
+                if token.type in (tokenize.ENDMARKER, tokenize.COMMENT):
+            	    continue
+                if (token.type == tokenize.OP) and (token.string == ';'):
```

Comment:
I tend to prefer the parentheses: I can never remember exactly which
operators have precedence, so explicit is better than implicit.
On 25 May 2015 5:21 pm, "Damián Avila" <notifications@github.com> wrote:

> In IPython/core/displayhook.py
> <https://github.com/ipython/ipython/pull/8457#discussion_r30998471>:
>
> >          try:
> > -            cell = self.shell.history_manager.input_hist_parsed[self.prompt_count]
> > -            return cell.rstrip().endswith(';')
> > +    	    for token in reversed(tokens):
> > +                if token.type in (tokenize.ENDMARKER, tokenize.COMMENT):
> > +            	    continue
> > +                if (token.type == tokenize.OP) and (token.string == ';'):
>
> I would remove all the parenthesis on this line...
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/ipython/ipython/pull/8457/files#r30998471>.
>